kong/plugins/ai-proxy/drivers/llama2.lua

```diff
@@ -0,0 +1,304 @@
+local _M = {}
+
+-- imports
+local cjson = require("cjson.safe")
+local split = require("pl.stringx").split
+local fmt = string.format
+local ai_shared = require("kong.plugins.ai-proxy.drivers.shared")
+local socket_url = require "socket.url"
+--
+
+-- parser built from model docs reference:
+-- https://huggingface.co/blog/llama2#how-to-prompt-llama-2
+local function messages_to_inst(messages)
+  local buf = require("string.buffer").new()
+  buf:reset()
+
+  for i, v in ipairs(messages) do
+    if i == 1 then
+      -- first, make the initial prompt
+        -- <s>[INST] <<SYS>>
+        -- {{ system_prompt }}
+        -- <</SYS>>
+      buf:put(fmt("<s>[INST] <<SYS>> %s <</SYS>>", v.content))
```

Comment:
The `:putf()` method has you covered for these

```suggestion
      buf:putf("<s>[INST] <<SYS>> %s <</SYS>>", v.content)
```